File: sample-objc/ARGEARSample/UI/UIComponent/Main/Bottom/Beauty/BeautyCollectionView.m
//
//  BeautyCollectionView.m
//  ARGEARSample
//
//  Created by Jihye on 24/09/2019.
//  Copyright © 2019 Seerslab. All rights reserved.
//

#import "BeautyCollectionView.h"

#define kBeautyCellNibName @"BeautyCollectionViewCell"
#define kBeautyCellIdentifier @"beautycell"

@implementation BeautyCollectionView

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    
    _iconNameArray = @[
        @"icVline",
        @"icFacewidth",
        @"icFacelength",
        @"icChinlength",
        @"icEyesize",
        @"icEyewidth",
        @"icNosenarrow",
        @"icAla",
        @"icNoselength",
        @"icMouthsize",
        @"icEyeback",
        @"icEyeangle",
        @"icLips",
        @"icSkin",
        @"icDarkcircle",
        @"icWrinkle",
    ];
    
    [self setDelegate:self];
    [self setDataSource:self];
    
    [self registerNib:[UINib nibWithNibName:kBeautyCellNibName bundle:nil] forCellWithReuseIdentifier:kBeautyCellIdentifier];
}

- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
    
    if ([_selectedIndexPath isEqual:indexPath]) {
        return;
    }
    
    [self setSelectedIndexPath:indexPath];
    [self reloadData];
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return [_iconNameArray count];
}

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
    return 1;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    BeautyCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:kBeautyCellIdentifier forIndexPath:indexPath];

    [[cell actionImage] setImage:[UIImage imageNamed:[self getIconName:[indexPath row]]]];
    
    return cell;
}

- (NSString *)getIconName:(NSInteger)index {
    
    NSMutableString *iconName = [NSMutableString stringWithFormat:@"%@", _iconNameArray[index]];
    
    if ([self tag] == ARGMediaRatio_16x9) {
        [iconName appendString:@"White"];
    } else {
        [iconName appendString:@"Black"];
    }
    
    if (index != [_selectedIndexPath row]) {
        [iconName appendString:@"Off"];
    }
    
    [iconName appendString:[NSString stringWithFormat:@"_%@", [@"" getLanguageCode]]];
    
    return iconName;
}

- (void)setRatio:(ARGMediaRatio)ratio {
    
    [self setTag:ratio];
    [self reloadData];
}

@end

File: sample-swift/ARGearSample/UI/UIComponent/Main/Bottom/Beauty/BeautyCollectionView.swift
//
//  BeautyCollectionView.swift
//  ARGearSample
//
//  Created by Jihye on 2020/01/30.
//  Copyright © 2020 Seerslab. All rights reserved.
//

import UIKit
import ARGear

class BeautyCollectionView: ARGBottomFunctionBaseCollectionView {

    let kBeautyCellNibName = "BeautyCollectionViewCell"
    let kBeautyCellIdentifier = "beautycell"
    
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.iconNames = [
            "icVline",
            "icFacewidth",
            "icFacelength",
            "icChinlength",
            "icEyesize",
            "icEyewidth",
            "icNosenarrow",
            "icAla",
            "icNoselength",
            "icMouthsize",
            "icEyeback",
            "icEyeangle",
            "icLips",
            "icSkin",
            "icDarkcircle",
            "icWrinkle"
        ]
        
        self.delegate = self
        self.dataSource = self
        
        self.register(UINib(nibName: kBeautyCellNibName, bundle: nil), forCellWithReuseIdentifier: kBeautyCellIdentifier)
    }
}

extension BeautyCollectionView: UICollectionViewDelegate {
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        if let selectedIndexPath = self.selectedIndexPath {
            if selectedIndexPath == indexPath {
                return
            }
        }
        
        self.selectedIndexPath = indexPath
        self.reloadData()
    }
}

extension BeautyCollectionView: UICollectionViewDataSource {
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return self.iconNames.count
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let beautyCell = collectionView.dequeueReusableCell(withReuseIdentifier: kBeautyCellIdentifier, for: indexPath) as! BeautyCollectionViewCell
        
        beautyCell.beautyImage.image = UIImage(named: self.getIconName(index: indexPath.row, withLanguageCode: true))
        
        return beautyCell
    }
}
